docs/cookbook/ddd_architecture/grid.md

```diff
@@ -0,0 +1,398 @@
+    {
+        $models = $this->queryBus->ask(new FindBooksQuery(
+            page: GridPageResolver::getCurrentPage($grid, $parameters),
+            itemsPerPage: GridPageResolver::getItemsPerPage($grid, $parameters),
+        ));
+
+        $data = [];
+        foreach ($models as $model) {
+            $data[] = BookResource::fromModel($model);
+        }
+
+        return new Pagerfanta(new FixedAdapter(count($models), $data));
+    }
+}
+```
+
+### Create the BookGrid
+
+We also need to create the `BookGrid`.
+
+```php
+// src/BookStore/Infrastructure/Sylius/Grid/BookGrid.php
+
+declare(strict_types=1);
+
+namespace App\BookStore\Infrastructure\Sylius\Grid;
+
+use App\BookStore\Infrastructure\Sylius\Resource\BookResource;
+use Sylius\Bundle\GridBundle\Builder\Action\CreateAction;
+use Sylius\Bundle\GridBundle\Builder\Action\DeleteAction;
+use Sylius\Bundle\GridBundle\Builder\Action\UpdateAction;
+use Sylius\Bundle\GridBundle\Builder\ActionGroup\BulkActionGroup;
+use Sylius\Bundle\GridBundle\Builder\ActionGroup\ItemActionGroup;
+use Sylius\Bundle\GridBundle\Builder\ActionGroup\MainActionGroup;
+use Sylius\Bundle\GridBundle\Builder\Field\StringField;
+use Sylius\Bundle\GridBundle\Builder\Field\TwigField;
+use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Bundle\GridBundle\Grid\AbstractGrid;
+use Sylius\Bundle\GridBundle\Grid\ResourceAwareGridInterface;
+
+final class BookGrid extends AbstractGrid implements ResourceAwareGridInterface
+{
+    public static function getName(): string
+    {
+        return self::class;
+    }
+
+    public function buildGrid(GridBuilderInterface $gridBuilder): void
+    {
+        $gridBuilder
+            ->setProvider(BookGridProvider::class) // The Grid provider we have just created
+            ->addField(
+                StringField::create('name')
+            )
+            ->addField(
+                StringField::create('author'),
+            )
+            ->addActionGroup(
+                MainActionGroup::create(
+                    CreateAction::create(),
+                ),
+            )
+            ->addActionGroup(
+                ItemActionGroup::create(
+                    UpdateAction::create(),
+                    DeleteAction::create(),
+                ),
+            )
+            ->addActionGroup(
+                BulkActionGroup::create(
+                    DeleteAction::create(),
+                ),
+            )
+        ;
+    }
+
+    public function getResourceClass(): string
+    {
+        return BookResource::class;
+    }
+}
+```
+
+### Add the grid on the Book Resource
+
+And then we create the "update" operation on our BookResource.
+
+```php
+// src/BookStore/Infrastructure/Sylius/Resource/BookResource.php
+
+// ...
+use App\BookStore\Infrastructure\Sylius\Grid\BookGrid;
+use Sylius\Resource\Metadata\Index;
+// ...
+
+#[AsResource(
+    // ...
+    formType: BookResourceType::class, // Define the form type for all your operations
```

Comment:
Both `BookResourceType` and `ResourceInterface` probably should be added to `use` block